plugins/module_utils/foreman_helper.py

```diff
@@ -467,6 +471,32 @@ def sanitize_entity_dict(entity_dict, name_map):
     return {value: entity_dict[key] for key, value in name_map.items() if key in entity_dict}
 
 
+def _entity_spec_helper(spec):
+    """Extend an entity spec by adding entries for all flat_names.
+    Extract ansible compatible argument_spec on the way.
+    """
+    entity_spec = {'id': {}}
+    argument_spec = {}
+    for key, value in spec.items():
+        entity_value = {}
+        argument_value = value.copy()
+        if 'flat_name' in argument_value:
+            flat_name = argument_value.pop('flat_name')
+            entity_value['flat_name'] = flat_name
+            entity_spec[flat_name] = {}
+
+        if argument_value.get('type') == 'entity':
+            argument_value.pop('type')
+            entity_value['type'] = 'entity'
+        elif value.get('type') == 'entity_list':
```

Comment:
```suggestion
        elif argument_value.get('type') == 'entity_list':
```

yes, they are copies, it just feels weird if one if branch uses `value` and the other `argument_value`